toolchain/semantics/testdata/empty.carbon

```diff
@@ -1,11 +1,7 @@
 // Part of the Carbon Language project, under the Apache License v2.0 with LLVM
 // Exceptions. See /LICENSE for license information.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-
-#include "toolchain/semantics/semantics_ir_for_test.h"
-
-namespace Carbon::Testing {
-
-llvm::Optional<SemanticsIR> SemanticsIRForTest::g_semantics;
-
-}  // namespace Carbon::Testing
+//
+// RUN: %{carbon} dump semantics-ir %s 2>&1 | %{FileCheck} %s
+// CHECK:{
+// CHECK:}
```

Comment:
Should we be using `CHECK-NEXT` here and throughout, or are you intending to allow there to be unmatched lines in the output?

I suppose we could use `--implicit-check-not={{$}}`, or something like that, in the `%{FileCheck}` substitution to check all lines are matched, rather than using `CHECK-NEXT` in all the tests.

Reply:
I'm hesitant to use CHECK-NEXT for verbosity. It's more text, but also still requires some kind of check to avoid unmatched output before and after. (and probably we should change all tests, right? this seems like an issue for explorer too)

`{{$}}` always fails with:

```
command line:1:22: error: CHECK-NOT: excluded string found in input
-implicit-check-not='{{$}}'
                     ^
<stdin>:1:1: note: found here
{
^
```

WDYT of `{{.+}}` instead? Is that doing what's intended?

This seems to only fail when I remove a check:

```
# command stderr:
command line:1:22: error: CHECK-NOT: excluded string found in input
-implicit-check-not='{{.+}}'
                     ^
<stdin>:2:1: note: found here
}
^
```

Also, dropping `--allow-unused-prefixes=false`; it seems to be the default now. I'd been checking because I thought this was what that did; I wish FileCheck had a more direct way of doing this.

Also, FileCheck -> FileCheckStrict to emphasize that there's a lot going on there now.

Reply:
Switched to `{{.}}` since that seems to be slightly more idiomatic, e.g.
https://github.com/llvm/llvm-project/blob/main/llvm/test/tools/llvm-strings/length.test

Reply:
Thanks, LG.